added use-case for on session payments via PaymentRequest (#48)
MewsSystems/gitbook-distributor-guide: distributor-api-v1/use-cases/on-session-payments.md

> # On session payments

Reviewer: We should add a link to this file and to to `distributor-api-v1/use-cases/README.md`  `https://github.com/MewsSystems/gitbook-distributor-guide/blob/master/SUMMARY.md` so it's visible in gitbook menu

Whenever a user create reservation there is an option to allow him pay directly. This needs to be correctly configured via
settlement rules on `RateGroup`. We provide this functionality only for `Rates` where the payment should be automatically

> taken immediately when the reservation group is created. In order to simplify the workflow we offer this functionality

Reviewer 2: The configuration is more complex. We are charging only rates that are settled right now and the method is set to charge credit card.

> in our payment gateway via a feature called "Payment request".

Reviewer: This paragraph is really similar to the one in `distributor-api-v1/use-cases/README.md`. Could some parts be unnecessarily duplicated and could some be moved to just one place?

## Creating a reservation group

> First step of the entire workflow is to [create a reservation group via our API](../operations.md#request-apibaseurlapidistributorv1reservationgroupscreate).

Reviewer: Maybe this link would be better - https://mews-systems.gitbook.io/distributor-guide/distributor-api-v1/operations#create-reservation-group ?

> The response contains a new field `PaymentRequestId` which we will use later on for a redirect to a payment gateway.

Reviewer 2: The response contains a new field `PaymentRequestId` that is used for redirection to a payment gateway.?

The payment gateway does support an optional query parameter `returnUrl` which will be used to return the user into

> your booking engine after successful payment or in case the user decide to abandon the flow. `returnUrl` value should

Reviewer: ```suggestion
your booking engine after successful payment or in case the user decides to abandon the flow. `returnUrl` value should
```

be a Base64 encoded absolute url e.g. `btoa(https://www.domain.tld/)`. After the user returns to your
booking engine you can verify the state of the payment request by calling [get reservation group API](../operations.md#get-reservation-group)

> with `Extent` for `PaymentRequests` and optionally for `Payments` where you want to validate.

Reviewer: ```suggestion
with `Extent` for `PaymentRequests` and optionally for `Payments` which you want to validate.
```

> Once the [Payment request State](../operations.md#payment-request-state) and optionally [Payment State](../operations.md#payment-state).

Reviewer: This is maybe missing a verb? I don't quite understand what it means

## Step by step workflow
1) [Create new reservation group](../operations.md#request-apibaseurlapidistributorv1reservationgroupscreate)
2) Get `PaymentRequestId` from the response

> 3) Create a `ReturnUrl` via encoding your url by `base64` e.g. `btoa('https://www.domain.tld/booking/completed/[ReservationGroupId]')`

Reviewer 2: I would remove the example of the javascript. It's use case. It's not developer documentation. 
or I would use just
e.g `btoa(...)`

Author: What about:
```
3) Create a `ReturnUrl` via encoding your url by `Base64` e.g. Javascript example would be `btoa(urlWhereUserShouldReturn)`
```

> 4) Redirect the user to the gateway on url `https://[MewsApplicationsBaseUrl]/navigator/payment-requests/detail/[PaymentRequestId]?returnUrl=[ReturnUrl]`

Reviewer: One thing for better understanding - maybe we could once more mention here where they should get `MewsApplicationsBaseUrl` and give them link to environments section in gitbook?

> 4) Redirect the user to the gateway on url `https://[MewsApplicationsBaseUrl]/navigator/payment-requests/detail/[PaymentRequestId]?returnUrl=[ReturnUrl]`. You can find out the `MewsApplicationBaseUrl` in [Environments](../environments.md) section.

Reviewer 2: `You can find out the `MewsApplicationBaseUrl` in [Environments](../environments.md) section` 

This makes it complicated. The developer can see the `MewsApplicationBaseUrl` several times. I would avoid it here. Or make it as link to guidelines.

> 5) After the user returns to the `ReturnUrl` in your booking engine you can verify the state by [reservation group detail](../operations.md#request-apibaseurlapidistributorv1reservationgroupsget) with a specified [Extent](../operations.md#reservation-group-get-extent)

Reviewer: ```suggestion
5) After the user returns to the `ReturnUrl` in your booking engine you can verify the state by using [reservation group detail](../operations.md#request-apibaseurlapidistributorv1reservationgroupsget) with a specified [Extent](../operations.md#reservation-group-get-extent)
```

6) Validate the state of the Payment request in the response

{% hint style="info" %}

> Till the Payment request enter its [finite state](../operations.md#payment-request-state) you can redirect the user again e.g. with "Try to pay again" button.

Reviewer: ```suggestion
Until the Payment request enters its [finite state](../operations.md#payment-request-state) you can redirect the user again e.g. with "Try to pay again" button.
```

> In case the Payment request expires or is canceled automatic settlement will take care of the payment, and it will try to charge the settlement amount as soon as possible.

Reviewer 2: Developer has no idea what the automatic settlement is.
Either to remove it or explain it what it means.
I would vote for removing it.
